modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/RuntimeMessageHandler.java

```diff
@@ -41,12 +48,51 @@
      * Add new error.
      * <p />
      * Error messages block server startup in strict mode.
+     *
+     * @deprecated since 11.3, use {@link #addMessage(Level, String, String, String, String)} or
+     *             {@link #addMessage(RuntimeMessage)} instead. O
      */
+    @Deprecated
     void addError(String message);
 
     /**
      * @return an unmodifiable {@link List} of error messages
      */
     List<String> getErrors();
 
+    void addMessage(RuntimeMessage message);
+
+    /**
+     * Adds the following message.
+     * <p>
+     * The source is a free string identifier allowing to distinguish between messages (and to clear them selectively if
+     * needed).
+     *
+     * @since 11.3
+     */
+    void addMessage(Level level, String message, Class<?> source);
+
+    /**
+     * Returns all messages, filtered by following predicate.
+     * <p>
+     * Predicate can be null, in which case no filtering will be done.
+     *
+     * @since 11.3
+     */
+    List<RuntimeMessage> getMessages(Predicate<RuntimeMessage> predicate);
+
+    /**
+     * Returns all messages strings, filtered by following predicate.
```

Comment:
Predicate could be `null`, like for `#getMessages`?

Reply:
Not anymore, see previous comments from Kevin: i removed this possibility

Reply:
So this need to be removed, right?
https://github.com/nuxeo/nuxeo/pull/4249/files#diff-ea74bf978f35cfe86fc57284c8fc0567R78

Reply:
Yes, thanks, i will review javadoc once we agree on the rest.